Dolores/src/main.rs

```diff
@@ -65,138 +62,114 @@ impl TypeMapKey for SnepContainer {
+            _ => return, // If Dolores doesn't recognize the prompt, don't do anything
         };
 
-        // Pass is used in order to not do anything if Dolores doesn't recognize anything
-        if responsemessage != String::from("Pass") {
-            let response = MessageBuilder::new().push(responsemessage).build();
+        let response = MessageBuilder::new().push(responsemessage).build();
 
-            if let Err(why) = msg.channel_id.say(&ctx.http, response).await {
-                println!(
-                    "The damn pony express failed to deliver the message! Goshdarnit!: {:?}",
-                    why
-                )
-            }
+        if let Err(why) = msg.channel_id.say(&ctx.http, response).await {
+            println!(
+                "The damn pony express failed to deliver the message! Goshdarnit!: {:?}",
+                why
+            )
         }
     }
 
     ///Logic that executes when the "Ready" event is received. The most important thing here is the logic for posting my species
     /// Because
     async fn ready(&self, ctx: Context, ready: Ready) {
         println!("{} is connected!", ready.user.name);
-        // My assumption is that once the await concludes we will continue to execute code 
+        // My assumption is that once the await concludes we will continue to execute code
         // The problem with awaiting a future in your current function is that once you've done that you are suspending execution until the future is complete
-        // As a result, if watch_westworld never finishes nothing else will be done from this function. period. ever. 
+        // As a result, if watch_westworld never finishes nothing else will be done from this function. period. ever.
         watch_westworld(&ctx).await;
 
         // Check to see if speciesupdateschannel exists, if it does not exist then create it
-        // Get all of the guilds channel ids 
+        // Get all of the guilds channel ids
         let channels = ctx
             .http
             .get_channels(511743033117638656)
             .await // stop executing the function until we get the results back, which is very useful
-            // If we don't have the permissions we need or can't connect to the API we should crash 
+            // If we don't have the permissions we need or can't connect to the API we should crash
             .expect("we were able to get the channels");
 
         // Set the specieschannelid so that we can update it once we find it
-        let mut specieschannelid = None;
-        for k in channels {
-            if k.name == "speciesupdates" {
-                specieschannelid = Some(k.id.0)
-            }
-        }
-         
-        
-        let mut map =  JsonMap::new();
+        let channel = channels
+            .into_iter()
+            .find(|channel| channel.name == "speciesupdates");
+        let specieschannelid = channel.map(|k| k.id.0);
```

Comment:
I did it this way because it exactly (modulo duplicate) matches the behavior of the previous loop through channels, and set `specieschannelid` to Some of the id of any that match the correct channel name.